File: Examen Parcial/20221309_EX1/Pregunta03/TransitSoft/TransitSoftPersistance/DAOImpl/VehiculoDAOImpl.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Data.Common;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;
using TransitSoft.DAO;
using TransitSoft.DAO.Util;
using TransitSoftModel;
using TransitSoftPersistance.DAO;
using static System.Collections.Specialized.BitVector32;

namespace TransitSoftPersistance.DAOImpl
{
    public class VehiculoDAOImpl : DAOImplBase, VehiculoDAO
    {
        private VehiculoDTO vehiculo;

        public VehiculoDAOImpl() : base("vehiculo")
        {
            this.retornarLlavePrimaria = true;
            this.vehiculo = null;
        }

        protected override void ConfigurarListaDeColumnas()
        {
            this.listaColumnas.Add(new Columna("id", true, true));
            this.listaColumnas.Add(new Columna("placa", false, false));
            this.listaColumnas.Add(new Columna("marca", false, false));
            this.listaColumnas.Add(new Columna("modelo", false, false));
            this.listaColumnas.Add(new Columna("anho", false, false));
        }

        protected override void IncluirValorDeParametrosParaInsercion()
        {
            AgregarParametro("@placa", this.vehiculo.Placa);
            AgregarParametro("@marca", this.vehiculo.Marca);
            AgregarParametro("@modelo", this.vehiculo.Modelo);
            AgregarParametro("@anho", this.vehiculo.Anho);
        }

        protected override void IncluirValorDeParametrosParaModificacion()
        {
            AgregarParametro("@id", this.vehiculo.Id);
            AgregarParametro("@placa", this.vehiculo.Placa);
            AgregarParametro("@marca", this.vehiculo.Marca);
            AgregarParametro("@modelo", this.vehiculo.Modelo);
            AgregarParametro("@anho", this.vehiculo.Anho);
        }

        protected override void IncluirValorDeParametrosParaEliminacion()
        {
            AgregarParametro("@id", this.vehiculo.Id);
        }

        protected override void IncluirValorDeParametrosParaObtenerPorId()
        {
            AgregarParametro("@id", this.vehiculo.Id);
        }

        protected override void InstanciarObjetoDelResultSet(DbDataReader lector)
        {
            this.vehiculo = new VehiculoDTO();
            this.vehiculo.Id = this.lector.GetInt32(0);
            this.vehiculo.Placa = this.lector.GetString(1);
            this.vehiculo.Marca = this.lector.GetString(2);
            this.vehiculo.Modelo = this.lector.GetString(3);
            this.vehiculo.Anho = this.lector.GetInt32(4);
            this.vehiculo.Propietario = new PropietarioDTO();
            this.vehiculo.Propietario.Id = this.lector.GetInt32(5);
        }

        protected override void LimpiarObjetoDelResultSet()
        {
            this.vehiculo = null;
        }

        protected override void AgregarObjetoALaLista(BindingList<Object> lista, DbDataReader lector)
        {
            this.InstanciarObjetoDelResultSet(lector);
            lista.Add(this.vehiculo);
        }

        //Metodos CRUD
        public int Insertar(VehiculoDTO vehiculo)
        {
            this.vehiculo = vehiculo;
            return base.Insertar();
        }

        public int Modificar(VehiculoDTO vehiculo)
        {
            this.vehiculo = vehiculo;
            return base.Modificar();
        }

        public int Eliminar(VehiculoDTO vehiculo)
        {
            this.vehiculo = vehiculo;
            return base.Eliminar();
        }

        public VehiculoDTO ObtenerPorId(int idVehiculo)
        {
            this.vehiculo = new VehiculoDTO();
            this.vehiculo.Id = idVehiculo;
            base.ObtenerPorId();
            return this.vehiculo;
        }

        public new BindingList<VehiculoDTO> ListarTodos()
        {
            BindingList<Object> lista;
            lista = base.ListarTodos();
            BindingList<VehiculoDTO> retorno = new BindingList<VehiculoDTO>();
            foreach (VehiculoDTO objecto in lista)
            {
                retorno.Add(objecto);
            }
            return retorno;
        }
    }
}

File: EX2/Conexion RESTful Transitsoft/TransitSoftPersistance/src/main/java/pe/edu/pucp/transitsoft/daoImp/VehiculoDAOImpl.java
package pe.edu.pucp.transitsoft.daoImp;

import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import pe.edu.pucp.transitsoft.dao.VehiculoDAO;
import pe.edu.pucp.transitsoft.daoImp.util.Columna;
import pe.edu.pucp.transitsoft.model.VehiculoDTO;

public class VehiculoDAOImpl extends DAOImplBase implements VehiculoDAO {
    
    private VehiculoDTO vehiculo;
    
    public VehiculoDAOImpl() {
        super("vehiculo");
        this.vehiculo = null;
        this.retornarLlavePrimaria = true;
    }
    
    @Override
    protected void configurarListaDeColumnas() {
        this.listaColumnas.add(new Columna("id", true, true));
        this.listaColumnas.add(new Columna("placa", false, false));
        this.listaColumnas.add(new Columna("marca", false, false));
        this.listaColumnas.add(new Columna("modelo", false, false));
        this.listaColumnas.add(new Columna("anho", false, false));
    }
    
    @Override
    protected void incluirValorDeParametrosParaInsercion() throws SQLException {
        this.statement.setString(1, this.vehiculo.getPlaca());
        this.statement.setString(2, this.vehiculo.getMarca());
        this.statement.setString(3, this.vehiculo.getModelo());
        this.statement.setInt(4, this.vehiculo.getAnho());
    }
    
    @Override
    protected void incluirValorDeParametrosParaModificacion() throws SQLException {        
        this.statement.setString(1, this.vehiculo.getPlaca());
        this.statement.setString(2, this.vehiculo.getMarca());
        this.statement.setString(3, this.vehiculo.getModelo());
        this.statement.setInt(4, this.vehiculo.getAnho());
        this.statement.setInt(5, this.vehiculo.getId());
    }
    
    @Override
    protected void incluirValorDeParametrosParaEliminacion() throws SQLException {
        this.statement.setInt(1, this.vehiculo.getId());
    }
    
    @Override
    protected void incluirValorDeParametrosParaObtenerPorId() throws SQLException {
        this.statement.setInt(1, this.vehiculo.getId());
    }
    
    @Override
    protected void instanciarObjetoDelResultSet() throws SQLException {
        this.vehiculo = new VehiculoDTO();
        this.vehiculo.setId(this.resultSet.getInt("id"));
        this.vehiculo.setPlaca(this.resultSet.getString("placa"));
        this.vehiculo.setMarca(this.resultSet.getString("marca"));
        this.vehiculo.setModelo(this.resultSet.getString("modelo"));
        this.vehiculo.setAnho(this.resultSet.getInt("anho"));
    }
    
    @Override
    protected void limpiarObjetoDelResultSet() {
        this.vehiculo = null;
    }
    
    @Override
    protected void agregarObjetoALaLista(List lista) throws SQLException{
        this.instanciarObjetoDelResultSet();
        lista.add(this.vehiculo);
    }
    
    @Override
    public Integer insertar(VehiculoDTO vehiculo) {        
        this.vehiculo = vehiculo;
        return super.insertar();
    }
    
    @Override
    public VehiculoDTO obtenerPorId(Integer id) {
        this.vehiculo = new VehiculoDTO();
        this.vehiculo.setId(id);
        super.obtenerPorId();
        return this.vehiculo;
    }
    
    @Override
    public ArrayList<VehiculoDTO> listarTodos() {
        return (ArrayList<VehiculoDTO>) super.listarTodos();
    }
    
    @Override
    public Integer modificar(VehiculoDTO vehiculo) {
        this.vehiculo = vehiculo;
        return super.modificar();
    }
    
    @Override
    public Integer eliminar(VehiculoDTO vehiculo) {
        this.vehiculo = vehiculo;
        return super.eliminar();
    }
    
}